[Bug]Deck options can't be opened

Thanks to developers.

###### Reproduction Steps

Open deck options

![IMG_20220319_141431.jpg](https://user-images.githubusercontent.com/59798823/159109751-bdf6b4e3-57b2-4fb9-885b-baca9faad436.jpg)
###### Expected Result

Enter the options interface.

###### Actual Result

Application crash，draw a blank

![Screenshot_2022-03-19-14-08-38-307_com.ichi2.anki.jpg](https://user-images.githubusercontent.com/59798823/159109592-1b7594c4-5ccf-49d7-8c01-9b0f5114b898.jpg)

###### Debug info
Refer to the [support page](https://ankidroid.org/docs/help.html) if you are unsure where to get the "debug info".

AnkiDroid Version = 2.16alpha52

Android Version = 11

Manufacturer = Xiaomi

Model = M2006J10C

Hardware = mt6885

Webview User Agent = Mozilla/5.0 (Linux; Android 11; M2006J10C Build/RP1A.200720.011; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/94.0.4606.85 Mobile Safari/537.36

ACRA UUID = d5adc2a0-6850-49bd-8875-6882e5d740bf

Scheduler = std2

Crash Reports Enabled = true

DatabaseV2 Enabled = true

###### Research
*Enter an [x] character to confirm the points below:*

- [x] I have read the [support page](https://ankidroid.org/docs/help.html) and am reporting a bug or enhancement request specific to AnkiDroid
- [x] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue
- [x] I have searched for similar existing issues here and on the user forum
- [x] (Optional) I have confirmed the issue is not resolved in the latest alpha release ([instructions](https://docs.ankidroid.org/manual.html#betaTesting))

```
java.lang.RuntimeException: Unable to start activity ComponentInfo{com.ichi2.anki/com.ichi2.anki.DeckOptions}: java.lang.NullPointerException: Attempt to invoke interface method 'java.lang.String android.content.SharedPreferences.getString(java.lang.String, java.lang.String)' on a null object reference
at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3623)
at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3795)
at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:85)
at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2227)
at android.os.Handler.dispatchMessage(Handler.java:106)
at android.os.Looper.loop(Looper.java:236)
at android.app.ActivityThread.main(ActivityThread.java:8130)
at java.lang.reflect.Method.invoke(Native Method)
at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:620)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1011)
Caused by: java.lang.NullPointerException: Attempt to invoke interface method 'java.lang.String android.content.SharedPreferences.getString(java.lang.String, java.lang.String)' on a null object reference
at com.ichi2.themes.Themes.themeFollowsSystem(Themes.kt:2)
at com.ichi2.themes.Themes.setThemeLegacy(Themes.kt:2)
at com.ichi2.anki.DeckOptions.onCreate(DeckOptions.java:1)
at android.app.Activity.performCreate(Activity.java:8142)
at android.app.Activity.performCreate(Activity.java:8114)
at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1309)
at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3596)
... 11 more
```

--- a/AnkiDroid/src/main/java/com/ichi2/anki/Preferences.kt
+++ b/AnkiDroid/src/main/java/com/ichi2/anki/Preferences.kt
@@ -838,7 +838,7 @@ class Preferences : AnkiActivity() {
                 appThemePref.entryValues = appThemesValues.sliceArray(1..appThemesValues.lastIndex)
             }
 
-            val followSystem = Themes.themeFollowsSystem(context)
+            val followSystem = Themes.themeFollowsSystem(requireContext())
             dayThemePref.isEnabled = followSystem
             nightThemePref.isEnabled = followSystem
 

--- a/AnkiDroid/src/main/java/com/ichi2/themes/Themes.kt
+++ b/AnkiDroid/src/main/java/com/ichi2/themes/Themes.kt
@@ -223,7 +223,7 @@ object Themes {
      * @return if user current selected theme is "Follow system"
      */
     @JvmStatic
-    fun themeFollowsSystem(context: Context?): Boolean {
+    fun themeFollowsSystem(context: Context): Boolean {
         val prefs = AnkiDroidApp.getSharedPrefs(context)
         val selectedAppTheme = prefs.getString(APP_THEME_KEY, FOLLOW_SYSTEM_MODE)
         return selectedAppTheme == FOLLOW_SYSTEM_MODE

--- a/AnkiDroid/src/main/java/com/ichi2/themes/Themes.kt
+++ b/AnkiDroid/src/main/java/com/ichi2/themes/Themes.kt
@@ -105,9 +105,12 @@ object Themes {
 
     @JvmStatic
     fun setThemeLegacy(context: Context) {
-        val prefs = AnkiDroidApp.getSharedPrefs(context.applicationContext)
-        if (themeFollowsSystem(context)) {
-            if (systemIsInNightMode(context)) {
+        // we need the applicationContext to obtain preferences, we can't do this with a regular
+        // Activity context since this is called before super.onCreate()
+        val applicationContext = context.applicationContext
+        val prefs = AnkiDroidApp.getSharedPrefs(applicationContext)
+        if (themeFollowsSystem(applicationContext)) {
+            if (systemIsInNightMode(applicationContext)) {
                 when (prefs.getString(NIGHT_THEME_KEY, NIGHT_BLACK_THEME)) {
                     NIGHT_BLACK_THEME -> context.setTheme(R.style.LegacyActionBarBlack)
                     NIGHT_DARK_THEME -> context.setTheme(R.style.LegacyActionBarDark)

